contrib/timelapse.lua

```diff
@@ -0,0 +1,463 @@
+  string_pref_write("filename_entry", "text")(filename_entry)
+
+  extra_data['images'] = images -- needed, to preserve images order
+  extra_data['tmp_dir'] = dt.configuration.tmp_dir .. '/'..MODULE_NAME .. '_' .. os.time()
+  extra_data['fps'] = framerates_selector.value
+  extra_data['res'] = extract_resolution(res_selector.value)
+  extra_data['codec'] = codec_selector.value
+  extra_data['img_ext'] = '.'..img_format.extension
+  local override_output = override_output_cb.value
+  local output_directory = auto_output_directory_btn.value and images[1].path or output_directory_chooser.value
+  local filename_mappings = {
+    date = os.date("%Y-%m-%d"),
+    time = os.date("%H-%M-%S"),
+    first_file = images[1].filename,
+    last_file = images[#images].filename
+  }
+  local output_extension = '.'..formats[format_selector.value]['extension']
+  local filename = format(filename_entry.text, filename_mappings)
+  local path = output_directory..OS_PATH_SEPARATOR..filename..output_extension
+  if not override_output then
+    path = df.create_unique_filename(path)
+  end
+
+  extra_data['output_file'] = path
+  extra_data['open_after_export'] = open_after_export_cb.value
+end
+
+local function export(extra_data)
+  local ffmpeg_path = df.check_if_bin_exists("ffmpeg")
+  if not ffmpeg_path then
+    dt.print_error("ffmpeg not found")
+    dt.print("ERROR - ffmpeg not found")
+    return
+  end
+  local dir = extra_data['tmp_dir']
+  local fps = extra_data['fps']
+  local res = extra_data['res']
+  local codec = extra_data['codec']
+  local img_ext = extra_data['img_ext']
+  local output_file = extra_data['output_file']
+  
+  local dir_create_result = mkdir(df.get_path(output_file))
+  if dir_create_result ~= 0 then return dir_create_result end
+
+  local cmd = ffmpeg_path..' -y -r '..fps..' -i '..dir..OS_PATH_SEPARATOR..'%d'..img_ext..' -s:v '..res..' -c:v '..codec..' -crf 18 -preset veryslow '..df.sanitize_filename(output_file)
+  return dsys.external_command(cmd), output_file
+end
+
+local function finalize_export(storage, images_table, extra_data)
+    local tmp_dir = extra_data['tmp_dir']
+    
+    dt.print(_('prepare merge process'))
+    
+    local result = mkdir(tmp_dir)
+    if result ~= 0 then dt.print(_('ERROR: cannot create temp directory')) end
+    
+    local images = extra_data['images']
+    -- rename all images to consecutive numbers
+    for i, file in pairs(images) do
+      local filename = images_table[file]
+      dt.print_error(filename, file.filename)
+      df.file_move(filename, tmp_dir .. OS_PATH_SEPARATOR .. i .. extra_data['img_ext'])
+    end
+    dt.print('Start video building...')
+    local result, path = export(extra_data)
+    if result ~= 0 then 
+      dt.print(_('ERROR: cannot build image, see console for more info')) 
+    else
+      dt.print(_('SUCCESS'))
+      if extra_data['open_after_export'] then
+        open_file(path)
+      end
+    end
+
+    rm(tmp_dir)
+end
+
```

Comment:
You could combine the above 3 functions into a function called process_timelapse and just do everything.  It would simplify your code because you wouldn't need any of the extra_table stuff.  You could just declare variables, set them and use them.  Put nil in place of the export and finalize_export functions and process_timelapse in place of the init_export function.

Reply:
I'm afraid I cannot do this:
1. first thing is that user might want to run/schedule a few bigger exports with different settings in one time. If I move reading of settings to the post-export phase, it will read the last settings, not the settings provided when export was initiated.
2. as far as I know finalize_export gets dictionary of exported images, that map `image file -> exported file name` but do **not** preserve original files order. This may be quite a big issue in video.